src/arch/src/lib.rs

```diff
@@ -100,7 +102,13 @@ impl FdtBuilder {
         self.rtc = Some((addr, size));
         self
     }
-
+    pub fn add_virito_device(&mut self, addr: u64, size: u64, irq: u32) -> &mut Self {
+        self.virtio_devices.push((addr, size, irq));
+        self
+    }
```

Comment:
Please add a blank line after each function definition. This is needed for all function you're defining.